packages/plugin-core/src/_extension.ts

```diff
@@ -899,6 +908,39 @@ export async function showLapsedUserMessage(assetUri: vscode.Uri) {
     });
 }
 
+export function shouldDisplayMissingDefaultConfigMessage(opts: {
+  ext: IDendronExtension;
+}): boolean {
+  const wsRoot = opts.ext.getDWorkspace().wsRoot;
+  const rawConfig = DConfig.getRaw(wsRoot);
+  const resp = ConfigUtils.detectMissingDefaults({ config: rawConfig });
+  return resp.data !== undefined && resp.data.needsBackfill;
+}
+
+export function showMissingDefaultConfigMessage(opts: {
+  ext: IDendronExtension;
+}) {
+  AnalyticsUtils.track(ConfigEvents.ShowMissingDefaultConfigMessage);
+  const ADD_CONFIG = "Add Missing Configuration";
+  vscode.window
+    .showInformationMessage(
+      "We have detected a missing default configuration. Would you like to add them to dendron.yml?",
+      ADD_CONFIG
+    )
+    .then(async (resp) => {
```

Comment:
If we make `showMissingDefaultConfigMessage` async, awaiting for it would block `_activate` until it's resolved. and by resolve, I mean it should be explicitly cleared by the user by removing it from the notification center or clicking on the X, or accepting the prompt. VSCode gives extensions a 15 second timer to display a notification and it will **hide** it. If it's hidden, it won't be resolved (Not sure if VSCode will go ahead and time them out if it's hanging)

Since it's returning `Promise<void>`, technically we don't really _need_ to wait for it. But I've learned today (along with the existence of the notification center UI) that the notification will be pushed away and hidden in the notification center almost instantly if we don't write it as a callback. I haven't looked into this deep enough to elaborate on what's going on here, but for the sake of consistency, I'll keep this in callback style like the other existing ones.

Reply:
Huh, that's weird. It's fine to keep it in that case.